docs/features/optin.md

```diff
@@ -0,0 +1,106 @@
+# Opt in to Enable Experimental Direct Controllers
+
+> `Feature State`: `alpha` as of version v1.123.1
+
+Config Connector can be configured to enable experimental versions of direct
+controllers for reconciling specific resources. This will allow users to test
+the new direct controller code for bug fixes or features not available in
+legacy controllers. It is possible to enable the direct controller code on a
+per-resource basis, by setting an annotation for each custom resource.
+
+We recommend only enabling the experimental direct controller if it is
+necessary to work around a legacy controller bug or to enable a new feature that
+is only available in the direct controller.
+
+
+## Enabling
+
+To enable the direct controller for a specific resource, update the resource and
+specify the annotation:
+
+```yaml
+metadata:
+  annotations:
+    alpha.cnrm.cloud.google.com/reconciler: "direct"
+```
+
+For example:
+
+```yaml
+apiVersion: sql.cnrm.cloud.google.com/v1beta1
+kind: SQLInstance
+metadata:
+  name: my-sqlinstance
+  annotations:
+    alpha.cnrm.cloud.google.com/reconciler: "direct"
+spec:
+  databaseVersion: POSTGRES_16
+  region: us-central1
+  settings:
+    tier: db-custom-1-3840
+```
+
+The only supported value for the annotation is "direct", to enable the
+direct controller.
+
+It's worth noting that it's (relatively) harmless to add the annotation to enable
```

Comment:
Not sure why we want to suggest adding the annotation for resources which don't yet support it. It seems like we are suggesting a mechanism to lay a land mine. 6 months later this seems likely to cause unexpected behavior when that controller is created and everyone has forgotten that a particular record had this annotation added.

Reply:
Maybe something more like "While adding an annotation to enable experimental direct controllers for resources that do not yet support experimental direct controllers should be harmless, we recommend against doing so. Adding such an annotation can be confusing and surprising when the functionality is added later."

Reply:
Good point, fixed!